src/item_location.cpp

```diff
@@ -255,7 +255,7 @@ class item_location::impl::item_on_map : public item_location::impl
             item *obj = target();
             item temp;
             if( target()->count_by_charges() ) {
-                temp = target()->split( qty );
+                temp = target()->split( qty,/*is_fake =*/true );
```

Comment:
imo instead of placing an overload on "innocent" function (as in, the function does what it's supposed to do) you can modify the way it's used weirdly instead, by creating a copy and splitting that;
```suggestion
                temp = item( *target() ).split( qty );
```

Reply:
Honestly, I implemented in this way at first during debugging, but later I chose to modify the function as not to introduce the overhead of copying large object.

Fair, you have the point. I will commit suggested changes. Thanks.

Reply:
if you want to save a copy then modifying Character::item_handling_cost to support a number of charges is a better target; put a default `..., int charges = -1` then in character.cpp line 6714 where it grabs the item::volume and that one has charges_in_vol to override charges, so you won't need to do any splitting at all

Reply:
Thank you, I'll look into it, maybe in another PR. This one will just be a quick-fix.